### PR TITLE
Use treemacs-magit package

### PR DESCRIPTION
**Pull request by jeffjewiss**
I thought this might be useful to setup and install when using both treemacs and magit. Cheers.

### CODE REVIEW

File: modules/ui/treemacs/config.el [hunk 1]
```diff
@@ -39,3 +39,7 @@
 
 (def-package! treemacs-projectile
   :after treemacs)
+
+(def-package! treemacs-magit
+  :when (and (featurep! :ui treemacs) (featurep! :tools magit))
```
Comment (hlissner) [inline modules/ui/treemacs/config.el]: Please remove this check for `:ui treemacs`. This is in the ui/treemacs module, none of which is evaluated if the containing module isn't enabled.

File: modules/ui/treemacs/packages.el [hunk 2]
```diff
@@ -5,3 +5,5 @@
 (when (featurep! :editor evil +everywhere)
   (package! treemacs-evil))
 (package! treemacs-projectile)
+(when (and (featurep! :ui treemacs) (featurep! :tools magit))
```
Comment (hlissner) [inline modules/ui/treemacs/packages.el]: Also remove this check for `:ui treemacs`. Same reason as above.

**Comment by jeffjewiss** [comment]
Whoops, sorry about the extra treemacs checks. Still new to emacs and lisp. Cheers, thanks!

**Comment by hlissner** [comment]
Not a problem! Thanks for the quick fixes (and thanks again for your help).

**Pull request merged**